### ISSUE TITLE
Next 13.4 breaks keystone

### ISSUE THREAD

**Issue by pc-erin** [keystonejs/keystone#8575]
Run `npm i && npm update`
Run `npm run dev`

The admin panel and console will show the following error:
`TypeError: Cannot read properties of null (reading 'useContext')`

If you run `npm ls next` you'll see that the version of next installed appears to be 13.4 instead of the 13.3 specified in the keystone dependencies.

It seems like next 13.4 causes some kind of problem for keystone.

My workaround so far is to add
```
"overrides": {
		"@keystone-6/core": {
			"next": "13.3.0"
		},
		"@keystone-6/auth": {
			"@keystone-6/core": {
				"next": "13.3.0"
			}
		},
		"@keystone-6/field-document": {
			"@keystone-6/core": {
				"next": "13.3.0"
			}
		}
	},
```
to my package.json file to lock the version of next to 13.3.0.

I'd suggest using exact versions in keystone's dependencies to prevent this sort of thing in the future.

**Comment by LunaTK** [comment]
as a temporary fix, you may use `resolution` field in package.json

```json
{
  "name": "my-app",
  "version": "1.0.2",
  "private": true,
  "scripts": {
    "dev": "keystone dev",
    "start": "keystone start",
    "build": "keystone build",
    "postinstall": "keystone build --no-ui --frozen"
  },
  "dependencies": {
    "@keystone-6/auth": "^7.0.0",
    "@keystone-6/core": "^5.0.0",
    "@keystone-6/fields-document": "^7.0.0",
    "typescript": "^4.9.5"
  },
  "resolutions": {
    "next": "~13.3.0"
  }
}
```

after running `yarn install` again, nextjs will be downgraded to 13.x

**Comment by dcousens** [comment]
Duplicate of https://github.com/keystonejs/keystone/issues/8553, but I'll leave this issue open until the newest version of Keystone is published.

**Comment by dcousens** [comment]
> I'd suggest using exact versions in keystone's dependencies to prevent this sort of thing in the future.

We will probably need to do this to be honest, but if we change it now, that may be a breaking change for some users.

**Comment by ciruz** [comment]
Damn I was thinking I did something wrong. @dcousens the thing is, if you setup keystone from the scratch, with the quick-start guide, it's broken as well: https://keystonejs.com/docs/getting-started#quick-start

So it's not only the update.

![image](https://github.com/keystonejs/keystone/assets/987947/e30a4f1a-0974-4755-a244-02576f14c0d1)

![image](https://github.com/keystonejs/keystone/assets/987947/ab8cb54f-c300-4acc-aa0d-4c0056ef751c)

**Comment by Durwinner** [comment]
Keystone requires a higher version of Next.js, specifically version 13.3 or above. If you encounter an error, please make sure to use at least this version. To fix the issue, you can add the following code to your package.json file:
```
"resolutions": {
  "next": "~13.3.0"
}
```
By adding this code, you're specifying a resolution to use Next.js version 13.3.0 or a compatible version. This can help resolve the problem you're facing. Make sure to save the package.json file after making this change.

**Comment by MohammadKurjieh** [comment]
This was resolved in #8568 please try using the latest version of keystone
